pytensor/tensor/rewriting/subtensor.py

```diff
@@ -381,49 +381,59 @@ def local_useless_slice(fgraph, node):
             return [node.inputs[0]]
 
 
-@register_infer_shape
+@register_useless
 @register_canonicalize
+@register_stabilize
 @register_specialize
 @node_rewriter([Subtensor])
 def local_replace_slice(fgraph, node):
     """
-    Remove Subtensor of the form:
-        1. X[0:-1:1] -> X[:-1]
-        2. X[0:-1] -> X[:-1]
-        3. X[:-1] -> X[:-1]
+    Rewrite Subtensor of the form:
+        1. X[0:-1:1] -> X[None:None:None]
+        2. X[0:-1:2] -> X[None:None:2]
+        3. X[3:-1] -> X[3:None:None]
 
     """
     idxs = get_idx_list(node.inputs, node.op.idx_list)
+    x = node.inputs[0]
 
     if not idxs:
-        return [node.inputs[0]]
+        return [x]
 
-    # last_slice = len(idxs)
+    new_idxs = list(idxs)
+    idx_flag = False
+    for dim, s in enumerate(new_idxs):
+        if not isinstance(s, slice):
+            continue
+
+        flag = False
+        start = s.start
+        stop = s.stop
+        step = s.step
+        if start is None or extract_constant(start, only_process_constants=True) == 0:
+            flag = True
+            start = None
 
-    for s in idxs[::-1]:
         if (
-            isinstance(s, slice)
-            and (
-                s.start is None
-                or extract_constant(s.start, only_process_constants=True) == 0
-            )
-            and extract_constant(s.stop, only_process_constants=True) == -1
-            and (
-                s.step is None
-                or extract_constant(s.step, only_process_constants=True) == 1
-            )
+            extract_constant(stop, only_process_constants=True) == -1
+            or extract_constant(stop, only_process_constants=True) == x.type.shape[dim]
         ):
-            # This does not work.
-            # I get the error that
-            # ```
-            # s.start = None
-            # AttributeError: readonly attribute
-            # ```
-            s.start = None
-            s.stop = -1
-            s.step = None
-        else:
-            break
+            flag = True
+            stop = None
+
+        if step is None or extract_constant(step, only_process_constants=True) == 1:
+            flag = True
```

Comment:
Better name for the flag?
```suggestion
            changed = True
```